Automattic/newspack-newsletters
# fix: do not auto subscribe to premium lists

### All Submissions:






### Changes proposed in this Pull Request:



This PR changes the behavior of Newsletters and Woo Memberships integration, aka Premium Newsletters.

Before this PR, when a user was granted with a Membership, they were automatically subscribed to all Lists that were locked behind that plan.

After this PR, they will no longer be automatically subscribed. Editor will then be able to offer the premium lists and only members of that plan will see them.

Also, it prevents a Fatal from throwing in an edge case when removing readers from lists.

### How to test the changes in this Pull Request:

1. Set up a Premium Newsletter behind a Membership that requires a product (see instructions at PCYsg-SZM-p2) 
2. Set up a checkout button to the product associated to the Membership plan
3. As an anonymous reader, buy the plan.
4. Confirm in the ESP that the reader was not added to the premium list
5. verify the reader`s account
6. Go to My Account -> Newsletters and confirm you can see the premium list listed there and you are able to subscribe
7. Confirm in the ESP that the subscription worked
8. As an admin, revoke the reader`s membership
9. In the ESP, confirm the user was removed from the Premium list

Also test with Subscription on the checkou modal

11. Add `define( 'NEWSPACK_ENABLE_POST_CHECKOUT_NEWSLETTER_SIGNUP', true );` to you wp-config
12. As a new anonymous reader, buy the plan via the checkout modal
13. Confirm that in the Thank you message, you can see the Premium listed offered for subscription
14. In the ESP confirm the reader was not added to the premium list until they opt-in

New tests after last commit

15. Checkout https://github.com/Automattic/newspack-plugin/pull/2778
16. Repeat steps 12 to 14 and confirm they still work
17. Now do the same thing but buying a product that will NOT grant you the membership
18. Confirm that in the checkout modal, the premium list is filtered out and does not show up

### Other information:
